ignite/handlers/lr_finder.py

```diff
@@ -118,13 +118,25 @@ def _run(
             )
 
         self.logger.debug(f"Running LR finder for {num_iter} iterations")
-        if start_lr is None:
-            start_lr = optimizer.param_groups[0]["lr"]
         # Initialize the proper learning rate policy
-        if step_mode.lower() == "exp":
-            start_lr_list = [start_lr] * len(optimizer.param_groups)
-            self._lr_schedule = LRScheduler(_ExponentialLR(optimizer, start_lr_list, end_lr, num_iter))
+        if step_mode.lower() == "exp":       
+            if start_lr is None:
+                start_lr_list = [optimizer.param_groups[i]["lr"] for i in range(len(optimizer.param_groups))]
+            elif isinstance(start_lr, float):
+                start_lr_list = [start_lr] * len(optimizer.param_groups)
+            else:
+                assert len(start_lr) == len(optimizer.param_groups)
+                start_lr_list = start_lr
+            if isinstance(end_lr, float):
+                end_lr_list = [end_lr] * len(optimizer.param_groups)
+            else:
+                assert len(end_lr) == len(optimizer.param_groups)
+                end_lr_list = end_lr
+            
+            self._lr_schedule = LRScheduler(_ExponentialLR(optimizer, start_lr_list, end_lr_list, num_iter))
         else:
+            if isinstance(start_lr, list) or isinstance(end_lr, list):
+                assert False, "THIS WON'T WORK"
```

Comment:
We can keep the code like that here. It was OK in the prototype but here we need : either raise an error `NotImplementedError` or 2) implement this case.